### ISSUE TITLE
cloudant 2.10.1+ requests of large docs slower than requests.get(...).json() or curl

### ISSUE THREAD

**Issue by dannyroberts** [cloudant/python-cloudant#414]
## Bug Description

Fetching some large documents from couch takes 20+s when the equivalent with `curl` or python `requests` takes 1s. It's a bug that was introduced in this PR: https://github.com/cloudant/python-cloudant/pull/412.

### 1. Steps to reproduce and the simplest code sample possible to demonstrate the issue

```
pip install cloudant
```

In python shell create a large document with non-ascii characters:
```
from cloudant.client import CouchDB
c = CouchDB('', '', url='http://localhost:5984', admin_party=True, connect=True)
# create large document with non ascii characters
x = c['my_database'].create_document({'_id': 'doc', 'value': u'好'  * 1000000})
```

Now test loading it with different versions of `cloudant` installed:

**2.10.0 is fast**
```
$ pip install cloudant==2.10.0
$ time python -c "from cloudant.client import CouchDB
c = CouchDB('', '', url='http://localhost:5984', admin_party=True, connect=True)
c['my_database']['doc']
"

real	0m0.365s
user	0m0.126s
sys	0m0.090s
```
**2.10.1 is slow**
```
$ pip install cloudant==2.10.1
$ time python -c "from cloudant.client import CouchDB
c = CouchDB('', '', url='http://localhost:5984', admin_party=True, connect=True)
c['my_database']['doc']
"

real	0m30.739s
user	0m29.667s
sys	0m0.314s
```

### 2. What you expected to happen

I expect it to be fast—about as fast as making the request directly with `requests.get(...).json()` or `curl`—and as fast as it is with `cloudant<=2.10.0`.

### 3. What actually happened

It takes 20+s to load, depending on the specific doc.

## Environment details
- Version(s) that are affected by this issue.
    > 2.10.1+
- Python version
    > 2.7

-------------

the cause is this PR: https://github.com/cloudant/python-cloudant/pull/412. The added slowness is concentrated entirely in `resp.text`, which does various things to infer the encoding, some of which are apparently expensive. The solution is to use `json.loads(resp.content.decode('utf-8'))` instead of `json.loads(resp.text)`.

**Comment by ricellis** [comment]
Thanks for reporting this. Agreed it appears the chardet can be particularly slow https://github.com/requests/requests/issues/2359 so we'll put the UTF-8 decoding in.

**Comment by dannyroberts** [comment]
Great, thanks for the quick turnaround!

What's the easiest way for me to track what version this get released in? Much appreciated!

**Comment by ricellis** [comment]
You can track the [2.next milestone](https://github.com/cloudant/python-cloudant/milestone/19) this issue is associated with. The milestone will get renamed to the real version before we release. Also if you follow the releases feed the changes included in [every release are summarized](https://github.com/cloudant/python-cloudant/releases).

At this point in time I expect it will be pretty much the only thing in a 2.10.2 release.

**Comment by smithsz** [comment]
@dannyroberts Thanks again for reporting this. The fix is now available in cloudant==2.10.2.
